packages/engine-core/src/common/utils/runtimeHeadersToHttpHeaders.ts

```diff
@@ -0,0 +1,21 @@
+import { QueryEngineRequestHeaders } from '../types/QueryEngine'
+
+/**
+ * Takes runtime data headers and turns it into QE HTTP headers
+ * @param headers to transform
+ * @returns
+ */
+export function runtimeHeadersToHttpHeaders(headers: QueryEngineRequestHeaders): Record<string, string | undefined> {
+  return Object.keys(headers).reduce((acc, runtimeHeaderKey) => {
+    let httpHeaderKey = runtimeHeaderKey
+
+    if (runtimeHeaderKey === 'transactionId') {
+      httpHeaderKey = 'X-transaction-id'
```

Comment:
Do we still need this header?

Reply:
For Data Proxy — not really, only for binary engine (this function is still used by both, it was actually extracted from `BinaryEngine.ts`). There's no harm in sending it for Data Proxy, but optimally we'd want to get rid of it because it should be ignored by the server. We'll change this after the release.